### PR TITLE
Update docs with additional steps for adding new types

### PR DESCRIPTION
**Pull request by dibyom**
# Changes

While adding new types for Conditionals in ( #1031 and #1130),  I found 
that there are quite a few places we need to update. Documenting this for future use!

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

### CODE REVIEW

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -270,7 +270,13 @@ To look at the logs for individual `TaskRuns` or `PipelineRuns`, see
 If you need to add a new CRD type, you will need to add:
 
 1. A yaml definition in [config/](./config)
-1. Add the type to the cluster roles in
-   [200-clusterrole.yaml](./config/200-clusterrole.yaml)
+1. Add the type to the cluster roles in:
```
Comment (afrittoli) [inline DEVELOPMENT.md]: This is very helpful, thank you!

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -270,7 +270,13 @@ To look at the logs for individual `TaskRuns` or `PipelineRuns`, see
...
+   - [200-clusterrole.yaml](./config/200-clusterrole.yaml)
+   - [clusterrole-aggregate-edit.yaml](./config/clusterrole-aggregate-edit.yaml)
+   - [clusterrole-aggregate-view.yaml](./config/clusterrole-aggregate-view.yaml)
+1. Add go structs for the types in [pkg/apis/pipelines/v1alpha1](./pkg/apis/pipeline/v1alpha1)
```
Comment (afrittoli) [inline DEVELOPMENT.md]: NIT: I'm not sure if we should include this, or only this. I guess the API types are the bare minimum, but perhaps we should include the [type]_validation and [type]_default modules as well; and perhaps the controller as well.

Comment (dibyom) [reply DEVELOPMENT.md]: hmmm, so in order to register with the webhook, the types have to be both `Defaultable` as well as `Validatable`. But probably worth calling this out explicitly.

**Pull request merged**